File: legacy-runtime/src/main/java/com/xiaomi/channel/commonutils/string/CloudCoder.java
package com.xiaomi.channel.commonutils.string;

import android.net.Uri;
import android.text.TextUtils;
import android.util.Log;
import java.io.UnsupportedEncodingException;
import java.security.InvalidAlgorithmParameterException;
import java.security.InvalidKeyException;
import java.security.InvalidParameterException;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.ArrayList;
import java.util.Map;
import java.util.TreeMap;
import javax.crypto.Cipher;
import javax.crypto.NoSuchPaddingException;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/string/CloudCoder.class */
public class CloudCoder {
    private static final String RC4_ALGORITHM_NAME = "RC4";
    private static final String TAG = "CloudCoder";

    public static String generateSignature(String str, String str2, Map<String, String> map, String str3) {
        if (TextUtils.isEmpty(str3)) {
            throw new InvalidParameterException("security is not nullable");
        }
        ArrayList<String> arrayList = new ArrayList<>();
        if (str != null) {
            arrayList.add(str.toUpperCase());
        }
        if (str2 != null) {
            arrayList.add(Uri.parse(str2).getEncodedPath());
        }
        if (map != null && !map.isEmpty()) {
            for (Map.Entry<String, String> entry : new TreeMap<>(map).entrySet()) {
                arrayList.add(String.format("%s=%s", entry.getKey(), entry.getValue()));
            }
        }
        arrayList.add(str3);
        boolean z = true;
        StringBuilder sb = new StringBuilder();
        for (String str4 : arrayList) {
            if (!z) {
                sb.append('&');
            }
            sb.append(str4);
            z = false;
        }
        return hash4SHA1(sb.toString());
    }

    public static String hash4SHA1(String str) {
        try {
            return String.valueOf(Base64Coder.encode(MessageDigest.getInstance("SHA1").digest(str.getBytes("UTF-8"))));
        } catch (UnsupportedEncodingException e) {
            Log.e(TAG, "CloudCoder.hash4SHA1", e);
            throw new IllegalStateException("failed to SHA1");
        } catch (NoSuchAlgorithmException e2) {
            Log.e(TAG, "CloudCoder.hash4SHA1", e2);
            throw new IllegalStateException("failed to SHA1");
        } catch (Exception e3) {
            Log.e(TAG, "CloudCoder.hash4SHA1", e3);
            throw new IllegalStateException("failed to SHA1");
        }
    }

    public static Cipher newAESCipher(String str, int i) {
        SecretKeySpec secretKeySpec = new SecretKeySpec(Base64Coder.decode(str), "AES");
        try {
            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
            cipher.init(i, secretKeySpec, new IvParameterSpec("0102030405060708".getBytes()));
            return cipher;
        } catch (InvalidAlgorithmParameterException e) {
            e.printStackTrace();
            return null;
        } catch (InvalidKeyException e2) {
            e2.printStackTrace();
            return null;
        } catch (NoSuchAlgorithmException e3) {
            e3.printStackTrace();
            return null;
        } catch (NoSuchPaddingException e4) {
            e4.printStackTrace();
            return null;
        }
    }

    public static Cipher newRC4Cipher(byte[] bArr, int i) {
        SecretKeySpec secretKeySpec = new SecretKeySpec(bArr, RC4_ALGORITHM_NAME);
        try {
            Cipher cipher = Cipher.getInstance(RC4_ALGORITHM_NAME);
            cipher.init(i, secretKeySpec);
            return cipher;
        } catch (InvalidKeyException e) {
            e.printStackTrace();
            return null;
        } catch (NoSuchAlgorithmException e2) {
            e2.printStackTrace();
            return null;
        } catch (NoSuchPaddingException e3) {
            e3.printStackTrace();
            return null;
        }
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/channel/commonutils/string/CloudCoder.kt
package com.xiaomi.channel.commonutils.string

import android.net.Uri
import android.text.TextUtils
import android.util.Log
import java.security.InvalidAlgorithmParameterException
import java.security.InvalidKeyException
import java.security.InvalidParameterException
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException
import java.util.*
import javax.crypto.Cipher
import javax.crypto.NoSuchPaddingException
import javax.crypto.spec.IvParameterSpec
import javax.crypto.spec.SecretKeySpec

object CloudCoder {
    private const val RC4_ALGORITHM_NAME = "RC4"
    private const val TAG = "CloudCoder"

    @JvmStatic
    fun generateSignature(
        method: String?,
        url: String?,
        params: Map<String, String>?,
        security: String
    ): String {
        if (TextUtils.isEmpty(security)) {
            throw InvalidParameterException("security is not nullable")
        }
        val arrayList = ArrayList<String>()
        if (method != null) {
            arrayList.add(method.uppercase())
        }
        if (url != null) {
            arrayList.add(Uri.parse(url).encodedPath ?: "")
        }
        if (params != null && params.isNotEmpty()) {
            for (entry in TreeMap(params).entries) {
                arrayList.add(String.format("%s=%s", entry.key, entry.value))
            }
        }
        arrayList.add(security)
        var z = true
        val sb = StringBuilder()
        for (str4 in arrayList) {
            if (!z) {
                sb.append('&')
            }
            sb.append(str4)
            z = false
        }
        return hash4SHA1(sb.toString())
    }

    @JvmStatic
    fun hash4SHA1(str: String): String {
        return try {
            String(
                Base64Coder.encode(
                    MessageDigest.getInstance("SHA1").digest(str.toByteArray(charset("UTF-8")))
                )
            )
        } catch (e: Exception) {
            Log.e(TAG, "CloudCoder.hash4SHA1", e)
            throw IllegalStateException("failed to SHA1")
        }
    }

    @JvmStatic
    fun newAESCipher(key: String, mode: Int): Cipher? {
        val secretKeySpec = SecretKeySpec(Base64Coder.decode(key), "AES")
        return try {
            val cipher = Cipher.getInstance("AES/CBC/PKCS5Padding")
            cipher.init(mode, secretKeySpec, IvParameterSpec("0102030405060708".toByteArray()))
            cipher
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }

    @JvmStatic
    fun newRC4Cipher(key: ByteArray, mode: Int): Cipher? {
        val secretKeySpec = SecretKeySpec(key, RC4_ALGORITHM_NAME)
        return try {
            val cipher = Cipher.getInstance(RC4_ALGORITHM_NAME)
            cipher.init(mode, secretKeySpec)
            cipher
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }
}
